### DISCUSSION TITLE
Is it possible to determine how many blocks a miner was eligible to mine in the past?

### DISCUSSION THREAD

**Discussion by neondragon** [filecoin-project/community#140] [Mining]
Filecoin miners often wonder if their 'luck' is normal, or if they have a hardware/software problem that has caused them to miss blocks that they could have mined.

1. Is it possible to determine which miners were eligible to mine blocks at each tipset, so that a network explorer website could show how many blocks each miner could have mined compared to how many they successfully mined?
2. Is it possible for a Filecoin implementation to have a command to allow a miner to view blocks in the past that they were eligible to mine and missed?

Or, is this information unavailable due to how the Expected Consensus algorithm works?

**Comment by funktafide** [comment]
I'm not sure about seeing eligibility. It would be nice to be able to see that. 

As far as looking at blocks that you were eligible for and miss for one reason or another, if you are running your miner with logs, you can grep the log for mined and it will show you the blocks you have mined or attempted and failed to mine.

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
there is a lotus issue open for that

https://github.com/filecoin-project/lotus/issues/6179

it is possible to replay all the elections, yes. but it is not implemented
